File: DSA/neetcode150/src/go/container_with_most_water.go
package main

func maxArea(height []int) int {
	l, r := 0, len(height)-1
	var result int
	for l < r {
		curr := (r - l) * min(height[l], height[r])
		result = max(result, curr)
		if height[r] > height[l] {
			l++
		} else {
			r--
		}
	}
	return result
}

File: docs/DSA/neetcode150/src/py/container_with_most_water.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        result = 0
        l, r = 0, len(height)-1
        while l < r:
            curr = (r-l)*min(height[l], height[r])
            result = max(result, curr)
            if height[r] > height[l]:
                l+=1
            else:
                r-=1
        return result